### PR TITLE
Improve RADOS GW support

### PR DESCRIPTION
**Pull request by cityofships**

### CODE REVIEW

File: roles/cephadm/templates/cluster.yml.j2 [hunk 1]
```diff
@@ -38,9 +38,13 @@ placement:
...
-  label: "rgw"
+  label: rgw
+{% if service.count_per_host is defined %}
+  count_per_host: {{ service.count_per_host }}
+{% endif %}
 {% if service.port is defined %}
-rgw_frontend_port: {{ service.port }}
+spec:
+  rgw_frontend_port: {{ service.port }}
 {% endif %}
```
Comment (mnasiadka) [inline roles/cephadm/templates/cluster.yml.j2:48]: What about the other options in the spec (https://docs.ceph.com/en/latest/cephadm/services/rgw/#service-specification)? We could just allow free form specification of additional settings?

Comment (cityofships) [reply roles/cephadm/templates/cluster.yml.j2:48]: Why not

Comment (cityofships) [reply roles/cephadm/templates/cluster.yml.j2:48]: What do we do with `service.port` then, should this be deprecated and replaced with `service.spec`?

**Review by mnasiadka — APPROVED** [review]

File: roles/cephadm/templates/cluster.yml.j2 [hunk 2]
```diff
@@ -42,9 +42,9 @@ placement:
 {% if service.count_per_host is defined %}
   count_per_host: {{ service.count_per_host }}
 {% endif %}
-{% if service.port is defined %}
```
Comment (markgoddard) [inline roles/cephadm/templates/cluster.yml.j2:42]: This is not backwards compatible, and will break anyone using `port`.

**Pull request merged**